compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -802,6 +806,47 @@ impl FirstSets {
     }
 }
 
+// Most `mbe::TokenTree`s are pre-existing in the matcher, but some are defined
+// implicitly, such as opening/closing delimiters and sequence repetition ops.
+// This type encapsulates both kinds. It implements `Clone` while avoiding the
+// need for `mbe::TokenTree` to implement `Clone`.
+#[derive(Clone, Debug)]
+enum TtHandle<'tt> {
+    Token(Token),
+    TtRef(&'tt mbe::TokenTree),
+}
```

Comment:
`TtHandle` is cloned directly within `TokenSet::add_all()`.

Reply:
I tried to do the `Cow` thing. It was a bit weird. First, I had to implement `ToOwned` for `mbe::TokenTree`. Second, it doesn't really match how `Cow` is supposed to work -- there is no possibility of a borrowed value `Borrowed` being converted to an `Owned` value, for example.

But the custom `Clone` was a good idea. I changed `TtHandle::Token` to contain a `mbe::TokenTree` instead of a `Token`, which got rid of the clumsiness for `TtHandle::get()`. And then I had to do a custom `Clone` for `TtHandle`, which was easy.